### PR TITLE
fix: make FilePicker work with latest dialogs

### LINKED ISSUE #2298: FilePicker: selectedNodes[0].extension is null
**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

**Describe the bug**
In the Results view when you open the file picker, the file/folder list loads forever. The browser console shows the following error:

```
TypeError: selectedNodes[0].extension is null
    picker Results.vue:742
    dialogButtons FilePicker-kaHdQqaz.mjs:956
    VueJS 4
    render15 FilePicker-kaHdQqaz.mjs:1047
    VueJS 7
vue.runtime.esm.js:3065:16
```

**To Reproduce**

1. Go to Results
2. Click on Create spreadsheet
3. See error

**Expected behavior**
File Picker should load normally

**Screenshots**
grafik

**Nextcloud (please complete the following information):**

-   Nextcloud-Version: 29/30
-   Forms-Version: 4.3.0-alpha5

**Additional context**
Probably a regression in `@nextcloud-dialogs` or `@nextcloud-vue`

### PR DESCRIPTION
**Pull request by Chartman123**
This fixes #2298 by adjusting the conditionnal logic in the buttonFactory

### CODE REVIEW

**Comment by Chartman123** [comment]
Change in .prettierignore is unrelated but necessary after latest workflow update for not having to reformat it after every update.

**Comment by Chartman123** [comment]
@susnux I could track it down that this error only happens when directories are allowed in the picker. When they are disallowed the error is gone, but of course you can't choose a directory for storage then...

File: .prettierignore [hunk 1]
```diff
@@ -3,6 +3,9 @@
 **/.svn
 **/.hg
 
+# Github workflows
+.github/
```
Comment (susnux) [inline .prettierignore:7]: I would prefer have it formatted on update (ideally formatted on the workflows repo).
But if not then at least:
```suggestion
.github/workflows/
```

File: src/views/Results.vue [hunk 2]
```diff
@@ -736,13 +736,40 @@ export default {
 				t('forms', 'Choose spreadsheet location'),
 			)
 				.setMultiSelect(false)
-				.allowDirectories()
+				.allowDirectories(true)
```
Comment (susnux) [inline src/views/Results.vue:739]: The looks like a bug in dialogs, thank you!

**Review by susnux — APPROVED** [review]
Thank you!

**Pull request merged**